component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/backupcode/BackupCodeAPIHandler.java

```diff
@@ -45,128 +41,96 @@
      * @throws BackupCodeException If an error occurred while generating the backup codes.
      */
-    public static Map<String, String> generateBackupCodes(String username, boolean refresh) throws BackupCodeException {
+    public static List<String> generateBackupCodes(String username) throws BackupCodeException {
 
-        String storedBackupCodes;
-        String generatedBackupCodes = "";
-        String tenantAwareUsername;
+        List<String> generatedBackupCodes = new ArrayList<>();
         String tenantDomain;
-        Map<String, String> claims = new HashMap<>();
         try {
-            tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
             UserRealm userRealm = BackupCodeUtil.getUserRealm(username);
             tenantDomain = MultitenantUtils.getTenantDomain(username);
-
             if (userRealm != null) {
-                Map<String, String> userClaimValues = userRealm.getUserStoreManager()
-                        .getUserClaimValues(tenantAwareUsername, new String[]{BACKUP_CODES_CLAIM}, null);
-                storedBackupCodes = userClaimValues.get(BACKUP_CODES_CLAIM);
-                if (StringUtils.isEmpty(storedBackupCodes) || refresh) {
-                    String plainBackupCodes = BackupCodeUtil.generateBackupCodes(tenantDomain);
-                    if (StringUtils.isNotEmpty(plainBackupCodes)) {
-                        generatedBackupCodes = BackupCodeUtil.encrypt(plainBackupCodes);
-                    }
-                } else {
-                    generatedBackupCodes = storedBackupCodes;
-                }
-                claims.put(BACKUP_CODES_CLAIM, generatedBackupCodes);
-                claims.put(BACKUP_CODES_ENABLED_CLAIM, "true");
+                generatedBackupCodes = BackupCodeUtil.generateBackupCodes(tenantDomain);
             }
         } catch (UserStoreException e) {
             throw new BackupCodeException(ERROR_CODE_ERROR_ACCESS_USER_REALM.getCode(),
                     String.format(ERROR_CODE_ERROR_ACCESS_USER_REALM.getMessage(), username, e));
-        } catch (CryptoException e) {
-            throw new BackupCodeException(ERROR_CODE_ERROR_ENCRYPT_BACKUP_CODE.getCode(),
-                    ERROR_CODE_ERROR_ENCRYPT_BACKUP_CODE.getMessage(), e);
         }
-        return claims;
+        return generatedBackupCodes;
     }
 
     /**
      * Update backup code claims and return backup codes for the user.
      *
-     * @param claims   Map with the backup code claims.
+     * @param generatedBackupCodes Generated backup codes as a comma separated string.
      * @param username Username of the user.
      * @return Backup codes for the user.
      * @throws BackupCodeException If an error occurred while updating backup codes.
      */
-    public static List<String> updateBackupCodes(Map<String, String> claims, String username)
+    public static List<String> updateBackupCodes(List<String> generatedBackupCodes, String username)
```

Comment:
Why do we need a public update method?

Reply:
addressed with [https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/6/commits/edff4471700af3eadb76c31edc00e6a7d59534ad](https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/6/commits/edff4471700af3eadb76c31edc00e6a7d59534ad)